vars/checkReleaseNotes.groovy

```diff
@@ -0,0 +1,118 @@
+/*
+ * Copyright OpenSearch Contributors
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * The OpenSearch Contributors require contributions made to
+ * this file be licensed under the Apache-2.0 license or a
+ * compatible open source license.
+ */
+import jenkins.ReleaseMetricsData
+import jenkins.ParseReleaseNotesMarkdownTable
+/**
+ * Library to check and notify missing release notes.
+ * @param Map args = [:] args A map of the following parameters
+ * @param args.version <required> - Release version. eg: 3.0.0
+ * @param args.dataTable <required> - Markdown data table file in the format: https://github.com/opensearch-project/opensearch-build/issues/3747#issuecomment-2704366641 eg: ./table.md
+ * @param args.action <optional> - Action to perform. Default is 'check'. Acceptable values are 'check' and 'notify'.
+ */
+void call(Map args = [:]) {
+    String action = args.action ?: 'check'
+    // Parameter check
+    validateParameters(args, action)
+    def version = args.version.tokenize('-')[0]
+
+    def parsedContent = new ParseReleaseNotesMarkdownTable(readFile(args.dataTable)).parseReleaseNotesMarkdownTableRows()
```

Comment:
Do we pass the complete table here or we use the link?

Reply:
Entire table as a file.